roles/edpm_ovs/tasks/update.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Check if ovs update is available
+  become: true
+  ansible.builtin.dnf:  # noqa: package-latest
+    name: "{{ edpm_ovs_packages }}"
+    state: latest
+  check_mode: true
+  register: ovs_update_check
+
+- name: Perform ovs update
+  when:
+    - ovs_update_check.rc == 0
+    - ovs_update_check.results | length > 0
+  become: true
+  block:
+    - name: Cache set ovs packages
+      ansible.builtin.include_tasks: download_cache.yml
```

Comment:
When running this as part of update seems we need to also enable state: latest in https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_ovs/tasks/download_cache.yml#L3 as without that next step seems can run with cache if present?

Reply:
I tested this before this change and dnf will handle update scenario correctly with the default `state` present.
E.g.
ovs-3.1 installed and ovs3.2 is requested. The cache directory will be updated to ovs3.2
ovs-3.1 installed and ovs3.1 is requested. The cache directory will stay unchanged and no update needed will be returned
ovs-3.1 installed and some ovs2.17 version is requested. Will fail with conflict version non zero rc code will be set. 

I also tried setting `latest` and the result is the same. Looking at the ansible code, they appear to be doing the same operation.
[ensure function](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L996)
[present](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L1021)
[latest](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L1027)

But for consistancy I can allow passing the state variable

Reply:
> I tested this before this change and dnf will handle update scenario correctly with the default `state` present. E.g. ovs-3.1 installed and ovs3.2 is requested. The cache directory will be updated to ovs3.2 ovs-3.1 installed and ovs3.1 is requested. The cache directory will stay unchanged and no update needed will be returned ovs-3.1 installed and some ovs2.17 version is requested. Will fail with conflict version non zero rc code will be set.
> 
> I also tried setting `latest` and the result is the same. Looking at the ansible code, they appear to be doing the same operation. [ensure function](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L996) [present](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L1021) [latest](https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L1027)
> 
It had failed for me with rdo-openvswitch-3.1 installed, update to rdo-openvswitch-3.2 succeeded, followed by rdo-openvswitch failed as download-cache didn't updated the cache, if this scenario works for you somehow i can cross check 
> But for consistancy I can allow passing the state variable